File: atcoder.jp/contests/abc212/tasks/abc212_b/main.rb
# frozen_string_literal: true

require 'set'

def weak1?(xs = [])
  Set.new(xs).size == 1
end

def weak2?(xs = [])
  return true if xs.size < 2

  x = xs[0]
  y = xs[1]
  (x + 1) % 10 == y && weak2?(xs[1..])
end

XS = gets.chomp.chars.map(&:to_i)

is_weak = weak1?(XS) || weak2?(XS)

puts is_weak ? 'Weak' : 'Strong'

File: atcoder.jp/contests/abc212/tasks/abc212_b/main.rs
use proconio::input;
use std::collections::HashSet;

fn main() {
    input! {
        xs: String,
    }

    let xs = xs.chars().map(|x| x.to_digit(10).unwrap()).collect();
    let is_weak = is_weak1(&xs) || is_weak2(&xs);

    println!("{}", if is_weak { "Weak" } else { "Strong" });
}

fn is_weak1(xs: &Vec<u32>) -> bool {
    let hs: HashSet<_> = xs.into_iter().collect();
    hs.len() == 1
}

fn is_weak2(xs: &[u32]) -> bool {
    if xs.len() < 2 {
        return true;
    }
    let x = xs[0];
    let y = xs[1];
    let xs = &xs[1..];
    (x + 1) % 10 == y && is_weak2(xs)
}
